tests/suites/test_suite_psa_crypto.function

```diff
@@ -6631,6 +6631,116 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_ECP_RESTARTABLE */
+void sign_hash_interruptible( int key_type_arg, data_t *key_data,
+                              int alg_arg, data_t *input_data,
+                              data_t *output_data, int max_ops )
+{
+    mbedtls_svc_key_id_t key = MBEDTLS_SVC_KEY_ID_INIT;
+    psa_key_type_t key_type = key_type_arg;
+    psa_algorithm_t alg = alg_arg;
+    size_t key_bits;
+    unsigned char *signature = NULL;
+    size_t signature_size;
+    size_t signature_length = 0xdeadbeef;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_status_t status = PSA_OPERATION_INCOMPLETE;
+    size_t num_ops = 0;
+    size_t num_ops_prior = 0;
+    size_t num_completes = 0;
+    size_t min_completes = 0;
+    size_t max_completes = 0;
+    psa_sign_hash_interruptible_operation_t operation =
+                                 psa_sign_hash_interruptible_operation_init( );
+
+    PSA_ASSERT( psa_crypto_init( ) );
+
+    psa_set_key_usage_flags( &attributes, PSA_KEY_USAGE_SIGN_HASH );
+    psa_set_key_algorithm( &attributes, alg );
+    psa_set_key_type( &attributes, key_type );
+
+    PSA_ASSERT( psa_import_key( &attributes, key_data->x, key_data->len,
+                                &key ) );
+    PSA_ASSERT( psa_get_key_attributes( key, &attributes ) );
+    key_bits = psa_get_key_bits( &attributes );
+
+    /* Allocate a buffer which has the size advertised by the
+     * library. */
+    signature_size = PSA_SIGN_OUTPUT_SIZE( key_type,
+                                           key_bits, alg );
+    TEST_ASSERT( signature_size != 0 );
+    TEST_LE_U( signature_size, PSA_SIGNATURE_MAX_SIZE );
+    ASSERT_ALLOC( signature, signature_size );
+
+    psa_interruptible_set_max_ops( max_ops );
+
+    if( max_ops == PSA_INTERRUPTIBLE_MAX_OPS_UNLIMITED )
+    {
+        min_completes = 1;
+        max_completes = 1;
+    }
+    else
+    {
+        min_completes = 2;
+        max_completes = PSA_INTERRUPTIBLE_MAX_OPS_UNLIMITED;
+    }
+
+    num_ops_prior = psa_sign_hash_get_num_ops( &operation );
+    TEST_ASSERT( num_ops_prior == 0 );
+
+    /* Start performing the signature. */
+    PSA_ASSERT( psa_sign_hash_start( &operation, key, alg,
+                                     input_data->x, input_data->len ) );
+
+    num_ops_prior = psa_sign_hash_get_num_ops( &operation );
+    TEST_ASSERT( num_ops_prior == 0 );
+
+    /* Continue performing the signature until complete. */
+    do
+    {
+        status = psa_sign_hash_complete( &operation, signature, signature_size,
+                                            &signature_length );
+
+        num_completes++;
+
+        if( status == PSA_SUCCESS || status == PSA_OPERATION_INCOMPLETE )
+        {
+            num_ops = psa_sign_hash_get_num_ops( &operation );
+            TEST_ASSERT( num_ops > num_ops_prior );
```

Comment:
Please add a note that we assume that `complete` always makes progress. It is the case in our implementation, but it is (currently) not mandated by the PSA specification.